### DISCUSSION TITLE
Getting error in lesson 2

### DISCUSSION THREAD

**Discussion by avikat** [smartcontractkit/full-blockchain-solidity-course-py#888] [Q&A] [answered]
Hi Community, I am geting error msg.sender.transfer ------ TypeError: "send" and "transfer" are only available for objects of type "address payable", not "address". please see the error in ss also
thank you
![Screenshot (288)](https://user-images.githubusercontent.com/24275333/151589433-60a2c45e-a93d-480f-b499-a9d80c079395.png)

**Comment by 00Amir00** [comment, marked as answer]
Solution: payable(msg.sender).transfer(address(this).balance);
In Solidity, there is a difference between a normal address and a payable address, so the correct way to send ETH to a sender would be payable(msg.sender).transfer(address(this).balance); -> This converts the normal address to a payable address.

**Reply by cromewar** [reply]
Does @00Amir00 solution works for you @avikat ?

**Reply by P1X37s** [reply]
Worked for me cromewar

**Reply by avikat** [reply]
i think this error came when i just started the course then i solved it i don't know how. Sorry for posting here any error

**Reply by P1X37s** [reply]
To clarify, swapping msg.sender.transfer(address(this).balance); for payable(msg.sender).transfer(address(this).balance); solved the problem for me.

**Reply by cromewar** [reply]
I'm marking this as solved by @00Amir00 thanks for the hint pal.
